[GHF] Small cleanup

malfet opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #77006
* #77005
* __->__ #77004

- Use double quotes for Python string literals (per https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#strings )
- Do not post comment of merge_on_green called with dry_run argument
- Dismantle pyramid of doom (and avoid infinite while) by specifying
  timeout as exit criteria
- Use `handle_exception` for revert workflow as well

malfet committed:
[GHF] Small cleanup

- Use double quotes for Python string literals (per https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#strings )
- Do not post comment of merge_on_green called with dry_run argument
- Dismantle pyramid of doom (and avoid infinite while) by specifying
  timeout as exit criteria
- Use `handle_exception` for revert workflow as well

[ghstack-poisoned]

malfet committed:
Update on "[GHF] Small cleanup"


- Use double quotes for Python string literals (per https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#strings )
- Do not post comment of merge_on_green called with dry_run argument
- Dismantle pyramid of doom (and avoid infinite while) by specifying
  timeout as exit criteria
- Use `handle_exception` for revert workflow as well

[ghstack-poisoned]

mehtanirav commented on .github/scripts/trymerge.py:
```diff
@@ -816,42 +816,47 @@ def merge_on_green(pr_num: int, repo: GitRepo, dry_run: bool = False) -> None:
     last_exception = ''
-    while True:
+    elapsed_time = 0.0
+    while elapsed_time < 400 * 60:
```
curious why we have such long (6+ hours) timeout. Does merging PR also run additional CI jobs?

malfet replied:
No idea, 6h limits sounds wrong to me (and the job would probably be killed earlier)
@zengk95 care to explain?

zengk95 replied:
I was thinking the scenario where
Person creates PR > Triggers CI > Immediately after, they write merge this > CI can potentially take 4-5 hours to run so this script spins for around 6 hours (once we make those tests mandatory)? which only gives us a 1 or 2 hour overhead. I chose 6 hours cause I think that's the default GHA timeout. 

Or the scenario where they retrigger CI by pushing another commit after like an hour or so and still want to land it when green.

mehtanirav commented on .github/scripts/trymerge.py:
```diff
@@ -816,42 +816,47 @@ def merge_on_green(pr_num: int, repo: GitRepo, dry_run: bool = False) -> None:
+    # Finally report timeout back
+    msg = "Merged timed out after 6 hours. Please contact the pytorch_dev_infra team."
+    msg += f"The last exception was: {last_exception}"
+    gh_post_comment(org, project, pr_num, msg, dry_run=dry_run)
```
We are already posting comment inside `handle_exception`, so it is not required here.

malfet replied:
Good point, removing. cc: @zengk95

mehtanirav reviewed (approved):
LGTM. some minor comments.

malfet committed:
Update on "[GHF] Small cleanup"


- Use double quotes for Python string literals (per https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#strings )
- Do not post comment of merge_on_green called with dry_run argument
- Dismantle pyramid of doom (and avoid infinite while) by specifying
  timeout as exit criteria
- Use `handle_exception` for revert workflow as well

[ghstack-poisoned]

malfet commented:
@pytorchbot merge on green this
